### PR TITLE
Do not send userId to registry when ODO_DEBUG_TELEMETRY_FILE is set

### LINKED ISSUE #5937: when ODO_DEBUG_TELEMETRY_FILE is set odo should not send telementry via devfile library
/kind bug

## How did you run odo exactly?
```
ODO_DEBUG_TELEMETRY_FILE=telementry.json odo init --name test --devfile go
```

## Actual behavior
odo telemetry data is correctly saved only to the `telementry.json`, but when odo calls `library.PullStackFromRegistry` it still sends userId.

## Expected behavior
If `ODO_DEBUG_TELEMETRY_FILE` is set then `PullStackFromRegistry` and any other external call that can send telemetry should behave as the telemetry is disabled.

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind bug

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #5937

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: pkg/segment/integrations.go [hunk 1]
```diff
@@ -27,6 +27,10 @@ func getTelemetryForDevfileRegistry() (registryLibrary.TelemetryData, error) {
 		return td, nil
 	}
 
+	if GetDebugTelemetryFile() != "" {
+		return td, nil
+	}
+
```
Comment (valaparthvi) [inline pkg/segment/integrations.go:33]: To make this faster, perhaps you can put this check on L19, before checking if telemetry is enabled.

File: pkg/segment/integrations.go [hunk 2]
```diff
@@ -23,6 +23,10 @@ func getTelemetryForDevfileRegistry() (registryLibrary.TelemetryData, error) {
 		return td, err
 	}
 
+	if GetDebugTelemetryFile() != "" {
```
Comment (valaparthvi) [inline pkg/segment/integrations.go:26]: I meant to suggest you can move this to L19, because initializing the preference client.

**Review by valaparthvi — APPROVED** [review]

File: pkg/segment/integrations.go [hunk 3]
```diff
@@ -17,6 +17,10 @@ func getTelemetryForDevfileRegistry() (registryLibrary.TelemetryData, error) {
 		Client: TelemetryClient,
 	}
 
+	if GetDebugTelemetryFile() != "" {
```
Comment (valaparthvi) [inline pkg/segment/integrations.go:20]: There's a unit test for this well which might need enhancing, can you check?

**Review by valaparthvi — APPROVED** [review]

**Comment by feloy** [comment]
/override ci/prow/v4.10-integration-e2e
Tests pass on IBM Cloud

**Pull request merged**